### PR TITLE
feat: implement event system

### PR DESCRIPTION
**Pull request by choffmann**
Implement an event system. For now, when a tree has been created, updated and deleted, it will send an event which will be subscribed to by the tree cluster to update its locations based on the changed tree.

### CODE REVIEW

File: internal/entities/events.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+type Event interface {
+	Type() EventType
+}
+
+const (
+	EventTypeUpdateTree        EventType = "update tree"
+	EventTypeCreateTree        EventType = "create tree"
+	EventTypeDeleteTree        EventType = "delete tree"
+	EventTypeUpdateTreeCluster EventType = "update tree cluster"
```
Comment (doriengr) [inline internal/entities/events.go:13]: Do we only have to listen on Update for a tree cluster?

File: internal/service/domain/treecluster/treecluster.go [hunk 2]
```diff
@@ -73,10 +107,15 @@ func (s *TreeClusterService) Create(ctx context.Context, createTc *domain.TreeCl
 		tc.Description = createTc.Description
 		tc.Trees = trees
 
-		if err = s.locator.UpdateCluster(ctx, tc); err != nil {
-			return false, handleError(err)
+		lat, long, region, err := s.getUpdatedLatLong(ctx, tc)
```
Comment (doriengr) [inline internal/service/domain/treecluster/treecluster.go:110]: Maybe put this in a function? From line 151 onwards is the same code

File: internal/service/service.go [hunk 3]
```diff
@@ -81,23 +81,7 @@ type RegionService interface {
 type TreeClusterService interface {
 	Service
 	CrudService[domain.TreeCluster, domain.TreeClusterCreate, domain.TreeClusterUpdate]
-}
-
-type GeoClusterLocator interface {
```
Comment (doriengr) [inline internal/service/service.go:86]: 🔥

**Review by doriengr — APPROVED** [review]
I am in love 😍

File: internal/entities/events.go (hunk 1 above)
Comment (choffmann) [reply internal/entities/events.go:13]: No, but I would implement each event if it is really needed

**Pull request merged**